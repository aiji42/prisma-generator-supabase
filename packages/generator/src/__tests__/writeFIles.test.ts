import { Project, SourceFile } from 'ts-morph'
import {
  writeImportsAndExports,
  writeOperationMapping,
  writePrepareFunction,
  writeRelationMapping,
  writeTableMapping,
} from '../utils/writeFiles'
import { GeneratorOptions } from '@prisma/generator-helper'
import {
  getSampleGenerator,
  getSampleDMMF,
} from './__fixtures__/getSampleSchema'

let file: SourceFile

const oldEnv = { ...process.env }

beforeEach(() => {
  const project = new Project()
  file = project.createSourceFile('./dummy.ts')

  process.env = {
    ...oldEnv,
    DATABASE_URL: 'postgresql://postgres:postgres@localhost:54322/postgres',
    SUPABASE_URL: 'this is SUPABASE_URL',
    SUPABASE_ANON_KEY: 'this is SUPABASE_ANON_KEY',
    SUPABASE_URL_CUSTOM: 'this is SUPABASE_URL_CUSTOM',
    SUPABASE_ANON_KEY_CUSTOM: 'this is SUPABASE_ANON_KEY_CUSTOM',
  }
})

afterEach(() => {
  process.env = { ...oldEnv }
})

test('writeImportsAndExports; with no fetchModule', async () => {
  const options = {
    generator: await getSampleGenerator('generator_minimum'),
  } as GeneratorOptions
  writeImportsAndExports(file, options)
  expect(file.getFullText()).toMatchSnapshot()
})

test('writeImportsAndExports; with fetchModule (browser)', async () => {
  const options = {
    generator: await getSampleGenerator('generator_with_browser_fetch'),
  } as GeneratorOptions
  writeImportsAndExports(file, options)
  expect(file.getFullText()).toMatchSnapshot()
})

test('writeImportsAndExports; with fetchModule (node-fetch)', async () => {
  const options = {
    generator: await getSampleGenerator('generator_with_node_fetch'),
  } as GeneratorOptions
  writeImportsAndExports(file, options)
  expect(file.getFullText()).toMatchSnapshot()
})

test('writeOperationMapping', async () => {
  const options = {
    dmmf: await getSampleDMMF(),
  } as GeneratorOptions

  writeOperationMapping(file, options)
  expect(file.getFullText()).toMatchSnapshot()
})

test('writeRelationMapping', async () => {
  const options = {
    dmmf: await getSampleDMMF(),
  } as GeneratorOptions

  writeRelationMapping(file, options)
  expect(file.getFullText()).toMatchSnapshot()
})

test('writeTableMapping', async () => {
  const options = {
    dmmf: await getSampleDMMF(),
  } as GeneratorOptions

  writeTableMapping(file, options)
  expect(file.getFullText()).toMatchSnapshot()
})

test('writePrepareFunction; with no environments', async () => {
  process.env = {
    ...oldEnv,
    DATABASE_URL: 'postgresql://postgres:postgres@localhost:54322/postgres',
  }
  const options = {
    generator: await getSampleGenerator('generator_minimum'),
  } as GeneratorOptions
  writePrepareFunction(file, options)
  expect(file.getFullText()).toMatchSnapshot()
})

test('writePrepareFunction; with environments', async () => {
  const options = {
    generator: await getSampleGenerator('generator_minimum'),
  } as GeneratorOptions
  writePrepareFunction(file, options)
  expect(file.getFullText()).toMatchSnapshot()
})

test('writePrepareFunction; with environments and specify keys', async () => {
  const options = {
    generator: await getSampleGenerator('generator_with_sb_envs'),
  } as GeneratorOptions
  writePrepareFunction(file, options)
  expect(file.getFullText()).toMatchSnapshot()
})
