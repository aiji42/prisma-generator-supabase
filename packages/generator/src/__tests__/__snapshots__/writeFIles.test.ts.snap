// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`writeImportsAndExports; with fetchModule (browser) 1`] = `
"import { prepare } from \\"@sb-prisma/client\\";

export { createClient, sb } from \\"@sb-prisma/client\\";
"
`;

exports[`writeImportsAndExports; with fetchModule (node-fetch) 1`] = `
"import fetch from \\"node-fetch\\";
import { prepare } from \\"@sb-prisma/client\\";

export { createClient, sb } from \\"@sb-prisma/client\\";
"
`;

exports[`writeImportsAndExports; with no fetchModule 1`] = `
"import { prepare } from \\"@sb-prisma/client\\";

export { createClient, sb } from \\"@sb-prisma/client\\";
"
`;

exports[`writeModels 1`] = `
"const models = {
  User: {
    name: \\"User\\",
    dbName: null,
    isGenerated: false,
    primaryKey: null,
    uniqueFields: [],
    uniqueIndexes: [],
    fields: {
      id: {
        name: \\"id\\",
        kind: \\"scalar\\",
        isList: false,
        isRequired: true,
        isUnique: false,
        isId: true,
        isReadOnly: false,
        type: \\"String\\",
        hasDefaultValue: true,
        default: { name: \\"uuid\\", args: [] },
        isGenerated: false,
        isUpdatedAt: false,
      },
      email: {
        name: \\"email\\",
        kind: \\"scalar\\",
        isList: false,
        isRequired: true,
        isUnique: true,
        isId: false,
        isReadOnly: false,
        type: \\"String\\",
        hasDefaultValue: false,
        isGenerated: false,
        isUpdatedAt: false,
      },
      name: {
        name: \\"name\\",
        kind: \\"scalar\\",
        isList: false,
        isRequired: false,
        isUnique: false,
        isId: false,
        isReadOnly: false,
        type: \\"String\\",
        hasDefaultValue: false,
        isGenerated: false,
        isUpdatedAt: false,
      },
      teamId: {
        name: \\"teamId\\",
        kind: \\"scalar\\",
        isList: false,
        isRequired: false,
        isUnique: false,
        isId: false,
        isReadOnly: true,
        type: \\"Int\\",
        hasDefaultValue: false,
        isGenerated: false,
        isUpdatedAt: false,
      },
      updatedAt: {
        name: \\"updatedAt\\",
        kind: \\"scalar\\",
        isList: false,
        isRequired: true,
        isUnique: false,
        isId: false,
        isReadOnly: false,
        type: \\"DateTime\\",
        hasDefaultValue: true,
        default: { name: \\"now\\", args: [] },
        isGenerated: false,
        isUpdatedAt: true,
      },
      Team: {
        name: \\"Team\\",
        kind: \\"object\\",
        isList: false,
        isRequired: false,
        isUnique: false,
        isId: false,
        isReadOnly: false,
        type: \\"Team\\",
        hasDefaultValue: false,
        relationName: \\"TeamToUser\\",
        relationFromFields: [\\"teamId\\"],
        relationToFields: [\\"id\\"],
        isGenerated: false,
        isUpdatedAt: false,
      },
    },
  },
  Team: {
    name: \\"Team\\",
    dbName: \\"teams\\",
    isGenerated: false,
    primaryKey: null,
    uniqueFields: [],
    uniqueIndexes: [],
    fields: {
      id: {
        name: \\"id\\",
        kind: \\"scalar\\",
        isList: false,
        isRequired: true,
        isUnique: false,
        isId: true,
        isReadOnly: false,
        type: \\"Int\\",
        hasDefaultValue: true,
        default: { name: \\"autoincrement\\", args: [] },
        isGenerated: false,
        isUpdatedAt: false,
      },
      name: {
        name: \\"name\\",
        kind: \\"scalar\\",
        isList: false,
        isRequired: true,
        isUnique: false,
        isId: false,
        isReadOnly: false,
        type: \\"String\\",
        hasDefaultValue: false,
        isGenerated: false,
        isUpdatedAt: false,
      },
      users: {
        name: \\"users\\",
        kind: \\"object\\",
        isList: true,
        isRequired: true,
        isUnique: false,
        isId: false,
        isReadOnly: false,
        type: \\"User\\",
        hasDefaultValue: false,
        relationName: \\"TeamToUser\\",
        relationFromFields: [],
        relationToFields: [],
        isGenerated: false,
        isUpdatedAt: false,
      },
    },
  },
};
"
`;

exports[`writeOperationMapping 1`] = `
"const operationMapping = {
  aggregateUser: {
    model: \\"User\\",
    method: \\"aggregate\\",
  },
  createManyUser: {
    model: \\"User\\",
    method: \\"createMany\\",
  },
  createOneUser: {
    model: \\"User\\",
    method: \\"createOne\\",
  },
  deleteManyUser: {
    model: \\"User\\",
    method: \\"deleteMany\\",
  },
  deleteOneUser: {
    model: \\"User\\",
    method: \\"deleteOne\\",
  },
  findFirstUser: {
    model: \\"User\\",
    method: \\"findFirst\\",
  },
  findManyUser: {
    model: \\"User\\",
    method: \\"findMany\\",
  },
  findUniqueUser: {
    model: \\"User\\",
    method: \\"findUnique\\",
  },
  groupByUser: {
    model: \\"User\\",
    method: \\"groupBy\\",
  },
  updateManyUser: {
    model: \\"User\\",
    method: \\"updateMany\\",
  },
  updateOneUser: {
    model: \\"User\\",
    method: \\"updateOne\\",
  },
  upsertOneUser: {
    model: \\"User\\",
    method: \\"upsertOne\\",
  },
  aggregateTeam: {
    model: \\"Team\\",
    method: \\"aggregate\\",
  },
  createManyTeam: {
    model: \\"Team\\",
    method: \\"createMany\\",
  },
  createOneTeam: {
    model: \\"Team\\",
    method: \\"createOne\\",
  },
  deleteManyTeam: {
    model: \\"Team\\",
    method: \\"deleteMany\\",
  },
  deleteOneTeam: {
    model: \\"Team\\",
    method: \\"deleteOne\\",
  },
  findFirstTeam: {
    model: \\"Team\\",
    method: \\"findFirst\\",
  },
  findManyTeam: {
    model: \\"Team\\",
    method: \\"findMany\\",
  },
  findUniqueTeam: {
    model: \\"Team\\",
    method: \\"findUnique\\",
  },
  groupByTeam: {
    model: \\"Team\\",
    method: \\"groupBy\\",
  },
  updateManyTeam: {
    model: \\"Team\\",
    method: \\"updateMany\\",
  },
  updateOneTeam: {
    model: \\"Team\\",
    method: \\"updateOne\\",
  },
  upsertOneTeam: {
    model: \\"Team\\",
    method: \\"upsertOne\\",
  },
};
"
`;

exports[`writePrepareFunction; not specify keys 1`] = `
"prepare({
    endpoint: process.env.SUPABASE_URL,apikey: process.env.SUPABASE_API_KEY,
    //@ts-ignore
    fetch,modelMap: {
        operationMapping,models,
    },
})
"
`;

exports[`writePrepareFunction; specify keys 1`] = `
"prepare({
    endpoint: process.env.SUPABASE_URL_CUSTOM,apikey: process.env.SUPABASE_API_KEY_CUSTOM,
    //@ts-ignore
    fetch,modelMap: {
        operationMapping,models,
    },
})
"
`;
